src/library/export/dlglibraryexport.h

```diff
@@ -0,0 +1,56 @@
+#ifdef __DJINTEROP__
+#pragma once
+
+#include <memory>
+
+#include <QDialog>
+#include <QLineEdit>
+#include <QListWidget>
+#include <QTreeWidget>
+#include <QWidget>
+
+#include "library/export/enginelibraryexportrequest.h"
+#include "preferences/usersettings.h"
+#include "util/parented_ptr.h"
+
+class TrackCollectionManager;
+
+namespace mixxx {
+
+// The DlgLibraryExport class is a UI window that gathers information from
+// the user about how they would like to export the Mixxx library.
+//
+// Currently, the dialog only supports exporting to the Engine Library format,
+// but in future it is expected that this dialog could be expanded to include
+// other formats, and generate different export signals accordingly.
```

Comment:
Thank you for the nice documentation. We are now using `///` for documentation comments so these are picked up by IDEs and Doxygen.

Reply:
Have now replaced with `///` in this header and all others introduced in the PR.